esmf_regrid/schemes.py

```diff
@@ -1,19 +1,315 @@
+    grid_cube : cube
+        A rectilinear instance of iris.cube.Cube that supplies the desired
+        horizontal grid definition.
+    mdtol : float, optional
+        Tolerance of missing data. The value returned in each element of the
+        returned cube's data array will be masked if the fraction of masked
+        data in the overlapping cells of the source cube exceeds mdtol. This
+        fraction is calculated based on the area of masked cells within each
+        target cell. mdtol=0 means no missing data is tolerated while mdtol=1
+        will mean the resulting element will be masked if and only if all the
+        overlapping cells of the source cube are masked. Defaults to 0.
+
+    Returns
+    -------
+    cube
+        A new iris.cube.Cube instance.
+
+    """
+    regrid_info = _regrid_rectilinear_to_rectilinear__prepare(src_cube, grid_cube)
+    result = _regrid_rectilinear_to_rectilinear__perform(src_cube, regrid_info, mdtol)
+    return result
+
+
 class ESMFAreaWeighted:
-    """TBD: public class docstring."""
+    """
+    A scheme which can be recognised by an iris cube's regrid method.
+
+    This class describes an area-weighted regridding scheme for regridding
+    between horizontal grids with separated X and Y coordinates. It uses
+    ESMF to be able to handle grids in different coordinate systems.
+    """
+
+    def __init__(self, mdtol=0):
+        """
+        Area-weighted scheme for regridding between rectilinear grids.
+
+        Parameters
+        ----------
+        mdtol : float
+            Tolerance of missing data. The value returned in each element of
+            the returned array will be masked if the fraction of missing data
+            exceeds mdtol. This fraction is calculated based on the area of
+            masked cells within each target cell. mdtol=0 means no masked
+            data is tolerated while mdtol=1 will mean the resulting element
+            will be masked if and only if all the overlapping elements of the
+            source grid are masked. Defaults to 0.
+
+        """
+        if not (0 <= mdtol <= 1):
+            msg = "Value for mdtol must be in range 0 - 1, got {}."
+            raise ValueError(msg.format(mdtol))
+        self.mdtol = mdtol
+
+    def __repr__(self):
+        """Return a representation of the class."""
+        return "ESMFAreaWeighted(mdtol={})".format(self.mdtol)
 
     def regridder(self, src_grid, tgt_grid):
-        """TBD: public method docstring."""
-        return _ESMFAreaWeightedRegridder(src_grid, tgt_grid)
+        """
+        Create regridder to perform regridding from source grid to target grid.
+
+        Parameters
+        ----------
+        src_grid : cube
+            The cube defining the source grid.
+        tgt_grid : cube
+            The cube defining the target grid.
+
+        Returns
+        -------
+        ESMFAreaWeightedRegridder
+            A callable with the interface:
+                `callable(cube)`
```

Comment:
`callable` is a built-in function in python, can we write this to be more explicit?

I assume you mean something like 
> Returns a ESMFAreaWeightedRegridder regridder object, that can be called `regridder(cube)` where `cube` is a ...